File: skills/letta-skills-letta/letta-api-client/examples/typescript/01_basic_client.ts
/**
 * Basic client initialization for Letta.
 *
 * Run: npx ts-node 01_basic_client.ts
 */
import { Letta } from "@letta-ai/letta-client";

// =============================================================================
// Option 1: Letta Cloud
// =============================================================================

const cloudClient = new Letta({
  apiKey: process.env.LETTA_API_KEY ?? "",
});

async function cloudExample() {
  // Verify connection by creating a simple agent
  const agent = await cloudClient.agents.create({
    name: "test-connection",
    model: "anthropic/claude-sonnet-4-5-20250929",
    embedding: "openai/text-embedding-3-small",
    memory_blocks: [{ label: "persona", value: "Test agent" }],
  });
  console.log(`Connected to Letta Cloud. Created agent: ${agent.id}`);
  
  // Clean up
  await cloudClient.agents.delete(agent.id);
  console.log("Deleted test agent.");
}

// =============================================================================
// Option 2: Self-Hosted (Docker)
// =============================================================================

const localClient = new Letta({
  baseURL: "http://localhost:8283",
});

async function localExample() {
  try {
    // Try to connect to local server
    const agent = await localClient.agents.create({
      name: "local-test",
      model: "anthropic/claude-sonnet-4-5-20250929",
      embedding: "openai/text-embedding-3-small",
      memory_blocks: [{ label: "persona", value: "Test" }],
    });
    console.log(`Connected to local server. Created: ${agent.id}`);
    await localClient.agents.delete(agent.id);
  } catch (e: unknown) {
    console.log(`Local server not available: ${e instanceof Error ? e.message : String(e)}`);
  }
}

// =============================================================================
// Run examples
// =============================================================================

async function main() {
  await cloudExample();
  await localExample();
}

main().catch(console.error);

File: skills/letta-skills-letta/letta-api-client/examples/python/01_basic_client.py
"""
Basic client initialization for Letta.

Run: python 01_basic_client.py
"""
import os
from letta_client import Letta

# =============================================================================
# Option 1: Letta Cloud
# =============================================================================
# Get your API key from https://app.letta.com/api-keys

cloud_client = Letta(api_key=os.getenv("LETTA_API_KEY"))

# Verify connection by listing agents
agents = cloud_client.agents.list(limit=5)
print(f"Connected to Letta Cloud. Found {len(list(agents))} agents.")


# =============================================================================
# Option 2: Self-Hosted (Docker)
# =============================================================================
# Start local server: docker run -p 8283:8283 letta/letta:latest

local_client = Letta(base_url="http://localhost:8283")

# Verify connection
try:
    agents = local_client.agents.list(limit=5)
    print(f"Connected to local server. Found {len(list(agents))} agents.")
except Exception as e:
    print(f"Local server not running: {e}")


# =============================================================================
# Option 3: Async Client
# =============================================================================
import asyncio
from letta_client import AsyncLetta

async def async_example():
    async_client = AsyncLetta(api_key=os.getenv("LETTA_API_KEY"))
    
    # Async operations
    agents = async_client.agents.list(limit=5)
    count = 0
    async for agent in agents:
        count += 1
    print(f"Async client found {count} agents.")

# asyncio.run(async_example())
